ChelseasFieldJournal/src/GUI/GuiComposerHelpers.cs

```diff
@@ -0,0 +1,836 @@
+                    key = "elementlistpicker";
+                }
+                int quantityButtons = elems.Length;
+                double lineWidth = 0.0;
+                for (int i = 0; i < elems.Length; i++)
+                {
+                    int index = i;
+                    if (lineWidth > (double)maxLineWidth)
+                    {
+                        startBounds.fixedX -= lineWidth;
+                        startBounds.fixedY += startBounds.fixedHeight + 5.0;
+                        lineWidth = 0.0;
+                    }
+                    GuiElement elem = Activator.CreateInstance(pickertype, new object[]
+                    {
+                        composer.Api,
+                        elems[i],
+                        startBounds.FlatCopy()
+                    }) as GuiElement;
+                    composer.AddInteractiveElement(elem, key + "-" + i.ToString());
+                    (composer[key + "-" + i.ToString()] as GuiElementElementListPickerBase<T>).handler = delegate (bool on)
+                    {
+                        if (on)
+                        {
+                            onToggle(index);
+                            for (int j = 0; j < quantityButtons; j++)
+                            {
+                                if (j != index)
+                                {
+                                    (composer[key + "-" + j.ToString()] as GuiElementElementListPickerBase<T>).SetValue(false);
+                                }
+                            }
+                            return;
+                        }
+                        (composer[key + "-" + index.ToString()] as GuiElementElementListPickerBase<T>).SetValue(true);
+                    };
+                    startBounds.fixedX += startBounds.fixedWidth + 5.0;
+                    lineWidth += startBounds.fixedWidth + 5.0;
+                }
+            }
+            return composer;
+        }
+
+        public static GuiComposer AddHorizontalTabs(this GuiComposer composer, GuiTab[] tabs, ElementBounds bounds, Action<int> onTabClicked, CairoFont font, CairoFont selectedFont, string key = null)
+        {
+            if (!composer.Composed)
+            {
+                composer.AddInteractiveElement(new GuiElementHorizontalTabs(composer.Api, tabs, font, selectedFont, bounds, onTabClicked), key);
+            }
+            return composer;
+        }
+
+        public static GuiElementHorizontalTabs GetHorizontalTabs(this GuiComposer composer, string key)
+        {
+            return (GuiElementHorizontalTabs)composer.GetElement(key);
+        }
+
+        public static GuiElementIconListPicker GetIconListPicker(this GuiComposer composer, string key)
+        {
+            return (GuiElementIconListPicker)composer.GetElement(key);
+        }
+
+        public static void IconListPickerSetValue(this GuiComposer composer, string key, int selectedIndex)
+        {
+            int i = 0;
```

Comment:
Like the function above, we can likely use Linq for this. Using a while loop can be very iffy. It's valid since we're checking for null on it, but it's still just concerning to see it in C#.